File: lab6/C++/Ryadki.cpp
#pragma once
#include "Header.h"

class Ryadki
{
protected: string str;
public:
    Ryadki(string line)
    {
        str = line;
    }
    string GetRyadok()
    {
        return str;
    }
    virtual int RyadokLength()
    {
        return str.length();
    }
    virtual void InsertSymbols()
    {
        string s = ", isn't it ?";
        str = str + s;
    }
};

File: RyadkiLibrary/Ryadki.cs
using System;

namespace RyadkiLibrary
{
    public class Ryadki
    {
        protected string str;
        public Ryadki(string line)
        {
            str = line;
        }
        public string GetRyadok
        {
            get { return str; }
        }
        public virtual int RyadokLength()
        {
            return str.Length;
        }
        public virtual void InsertSymbols()
        {
            string s = ", isn't it ?";
            str = str + s;
        }
    }
}
